### PR TITLE
Hod/version mismatch

### PR DESCRIPTION
**Pull request by hod-alpert**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/29431

## Description
Failing the build in case branch's latest version is lower than the latest version in production bucket.

## Must have
- [x] Tests

### CODE REVIEW

File: Tests/Marketplace/Tests/marketplace_services_test.py [hunk 1]
```diff
@@ -456,6 +456,62 @@ def test_prepare_release_notes_upgrade_version_dup(self, mocker, dummy_pack):
...
+            },
+            "2.0.0": {
+                "releaseNotes": "Second release notes",
+                "displayName": "2.0.0",
+                "released": "2020-06-05T13:39:33Z"
+            }
+        }
+        branch_latest_version = '1.9.9'
+        try:
+            Pack.assert_production_bucket_version_matches_release_notes_version(dummy_pack,
```
Comment (Itay4) [inline Tests/Marketplace/Tests/marketplace_services_test.py:508]: better to use `pytest.raises` here and verify expected exception

**Comment by Itay4** [comment]
@hod-alpert how did we test this? do we have builds we triggered against these changes?

**Comment by hod-alpert** [comment]
> @hod-alpert how did we test this? do we have builds we triggered against these changes?

Tried yesterday and could not reproduce, I'll try again today

**Comment by hod-alpert** [comment]
@Itay4 This is a build that reproduces the issue - https://app.circleci.com/pipelines/github/demisto/content/39207/workflows/5fe33d52-ca32-4fbc-80b5-8144c5eefdc7
And this is a build with the fix - https://app.circleci.com/pipelines/github/demisto/content/39209/workflows/f57a76f6-66a8-4088-a807-920506c4982d

File: Tests/Marketplace/Tests/marketplace_services_test.py [hunk 2]
```diff
@@ -456,6 +456,59 @@ def test_prepare_release_notes_upgrade_version_dup(self, mocker, dummy_pack):
...
+                "released": "2020-05-05T13:39:33Z"
+            },
+            "2.0.0": {
+                "releaseNotes": "Second release notes",
+                "displayName": "2.0.0",
+                "released": "2020-06-05T13:39:33Z"
+            }
+        }
+        branch_latest_version = '1.9.9'
+        with pytest.raises(AssertionError):
```
Comment (Itay4) [inline Tests/Marketplace/Tests/marketplace_services_test.py:507]: lets also assert the exception value is as expected

**Comment by hod-alpert** [comment]
@Itay4 Another build link - https://app.circleci.com/pipelines/github/demisto/content/39240/workflows/0c8638ec-2a3e-4279-9637-4196bec20c24/jobs/169508

**Review by Itay4 — APPROVED** [review]

**Pull request merged**